pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -278,11 +278,40 @@ static int is_simple_frame(PyObject *obj) {
 }
 
 static npy_int64 get_long_attr(PyObject *o, const char *attr) {
+    // NB we are implicitly assuming that o is a Timedelta or Timestamp, or NaT
```

Comment:
that would entail a bunch more boilerplate in the ujson.c file i think.  doing the validation on the ".value" attribute should be enough.

Reply:
Where is that done? The problem with our C code is that we have too many assumptions around object types, so refactoring is a nightmare. When our assumptions don't hold true we get undefined behavior and segfaults

Reply:
im not sure i understand the question.  The .value lookup is the PyObject_GetAttrString on L284.  And both the places where this is called from are after hasattr checks.

In the past weve discussed porting this module to cython piecemeal and that never really panned out.  Might be worth trying an all-at-once one of these days.

Reply:
Ah OK I see your point - didn't notice that in the diff. The duck typed approach would be to make sure value isn't NULL on L284. Otherwise right now things probably segfault.

As far as porting to Cython...that would be a ton of work. Not sure that would make things easier either with the way ultrajson works

Reply:
I opened #49756 to track this separately from this PR